presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/KinesisColumnHandle.java

```diff
@@ -0,0 +1,168 @@
+    private final boolean hidden;
+    private final boolean internal;
+
+    @JsonCreator
+    public KinesisColumnHandle(
+            @JsonProperty("ordinalPosition") int ordinalPosition,
+            @JsonProperty("name") String name,
+            @JsonProperty("type") Type type,
+            @JsonProperty("mapping") String mapping,
+            @JsonProperty("dataFormat") String dataFormat,
+            @JsonProperty("formatHint") String formatHint,
+            @JsonProperty("hidden") boolean hidden,
+            @JsonProperty("internal") boolean internal)
+    {
+        this.ordinalPosition = ordinalPosition;
+        this.name = requireNonNull(name, "name is null");
+        this.type = requireNonNull(type, "type is null");
+        this.mapping = mapping;
+        this.dataFormat = dataFormat;
+        this.formatHint = formatHint;
+        this.hidden = hidden;
+        this.internal = internal;
+    }
+
+    @JsonProperty
+    public int getOrdinalPosition()
+    {
+        return ordinalPosition;
+    }
+
+    @JsonProperty
+    public String getName()
+    {
+        return name;
+    }
+
+    @JsonProperty
+    public Type getType()
+    {
+        return type;
+    }
+
+    @JsonProperty
+    public String getMapping()
+    {
+        return mapping;
+    }
+
+    @Nullable
+    @JsonProperty
+    public String getDataFormat()
+    {
+        return dataFormat;
+    }
+
+    @JsonProperty
+    public String getFormatHint()
+    {
+        return formatHint;
+    }
+
+    @JsonProperty
+    public boolean isHidden()
+    {
+        return hidden;
+    }
+
+    @JsonProperty
+    public boolean isInternal()
+    {
+        return internal;
+    }
+
+    ColumnMetadata getColumnMetadata()
+    {
+        return new ColumnMetadata(name, type, null, hidden); // name/type/comment/hidden
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(ordinalPosition, name, type, mapping, dataFormat, formatHint, hidden, internal);
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        KinesisColumnHandle other = (KinesisColumnHandle) obj;
+        return Objects.equals(this.ordinalPosition, other.ordinalPosition) &&
+                Objects.equals(this.name, other.name) &&
+                Objects.equals(this.type, other.type) &&
+                Objects.equals(this.mapping, other.mapping) &&
+                Objects.equals(this.dataFormat, other.dataFormat) &&
+                Objects.equals(this.formatHint, other.formatHint) &&
+                Objects.equals(this.hidden, other.hidden) &&
+                Objects.equals(this.internal, other.internal);
+    }
+
+    @Override
+    public int compareTo(KinesisColumnHandle otherHandle)
```

Comment:
This class is `Comparable`, but the implementation is not consistent with `equals`. This is fragile and can produce surprising results when elements of this class are added to some collection implementations, so it's generally discouraged. It we need this behavior, let's move it to an independent comparator class (maybe call it `ByOrdinalComparator`).

Reply:
RedisColumnHandle seems to have similar implementation, removed this for now as all tests are passing without implementing Comparable interface too.